src/internal.rs

```diff
@@ -410,9 +410,8 @@ pub struct Rav1dFrameContext_frame_thread {
     // indexed using (t->by >> 1) * (f->b4_stride >> 1) + (t->bx >> 1)
     pub pal: AlignedVec64<[[u16; 8]; 3]>, /* [3 plane][8 idx] */
     // iterated over inside tile state
-    pub pal_idx: *mut u8,
+    pub pal_idx: AlignedVec64<u8>,
```

Comment:
Looking into this more, I think we can store a `Vec<Align64<[u8; 64]>>`.  We allocate in multiples of `128 * 128 / 4 = 4096`, and `pal_idx` also sometimes comes from `[u8; 8192]`.  So both are divisible by `64`.  And when `pal_idx` is sliced in `fn rav1d_recon_b_intra`, it's done by `cbw4 * cbh4 * 16` and `bw4 * bh4 * 16`.  I believe the `4` in those variables may mean it's a multiple of 4 (@fbossen, is that right at all?).  Regardless, running the tests shows that all of those offsets are divisible by `64`.  So we can do the slice on the `Vec<Align64<[u8; 64]>>`, and then pass `&[Align64<[u8; 64]>]` all the way to the asm boundary (which we can upgrade to `*const Align64<[u8; 64]>`).  And `fn pal_pred_rust` can be rewritten to work with `idx: &[Align64<[u8; 64]>]` instead, such as by using the `fn flatten` I added recently.

Reply:
Let me take a look at how this would work out.